src/main/java/com/cedarsoftware/util/io/JsonReader.java

```diff
@@ -542,18 +542,20 @@ public JsonReader(InputStream inp, Map<String, Object> optionalArgs)
     public JsonReader(String inp, Map<String, Object> optionalArgs)
     {
         initializeFromArgs(optionalArgs);
-
-        try {
-            input = new FastPushbackBytesReader(inp.getBytes(Charset.forName("UTF-8")));
-        } catch (UnsupportedCharsetException e) {
+        try
+        {
+            input = new FastPushbackBufferedReader(new InputStreamReader(new ByteArrayInputStream(inp.getBytes("UTF-8"))));
```

Comment:
The FastPushbackBytesReader did not properly support UTF-8.  Because it was processing byte-by-byte, it was appending invalid code points to the StringBuilder.  This causes an IllegalArgumentException.

As much of the optimization was kept as possible, for example, the refactoring of the PushbackReader into an interface, as well as the signature of the new public APIs on JsonReader.  In order to maintain a properly functioning version of json-io, we cannot use the existing FastPushbackBytesReader, hence it was removed. 

If you can correct the FastPushbackBytesReader to be UTF-8 (UTF-*) then please make a PR for that.  Also, make sure appropriate tests are added.  A new test was added today in TestString.groovy that processes a string with a French character in it.  This test illustrates the problem with the FastPushbackBytesReader as it stands.